component/src/test/java/org/wso2/extension/siddhi/io/jms/source/JMSSourceTestCase.java

```diff
@@ -255,7 +262,174 @@ public void receive(long timeStamp, Event[] inEvents, Event[] removeEvents) {
                 }
             }
         });
+        Collection<List<Source>> sources = executionPlanRuntime.getSources();
+        executionPlanRuntime.start();
+        // pause
+        sources.forEach(e -> e.forEach(Source::pause));
+        // send few events
+        messageList.clear();
+        messageList.add("<events>\n"
+                + "    <event>\n"
+                + "        <name>John</name>\n"
+                + "        <age>22</age>\n"
+                + "        <country>US</country>\n"
+                + "    </event>\n"
+                + "</events>");
+        Thread.sleep(1000);
+        // publishing events
+        publishEvents("DAS_JMS_TEST", null, "activemq", "text", messageList);
+
+        //resume
+        sources.forEach(e -> e.forEach(Source::resume));
+        // send few more events
+        messageList.clear();
+        messageList.add("<events>\n"
+                + "    <event>\n"
+                + "        <name>Johny</name>\n"
+                + "        <age>22</age>\n"
+                + "        <country>US</country>\n"
+                + "    </event>\n"
+                + "</events>");
+        // publishing events
+        publishEvents("DAS_JMS_TEST", null, "activemq", "text", messageList);
+
+        List<String> expected = new ArrayList<>(1);
+        expected.add("John");
+        expected.add("Johny");
+        SiddhiTestHelper.waitForEvents(waitTime, 2, eventCount, timeout);
+        Assert.assertEquals(eventCount.intValue(), 2, "Event count should be equal to two.");
+        AssertJUnit.assertEquals("JMS Source expected input not received", expected, receivedEventNameList);
+        siddhiManager.shutdown();
+    }
+
+    @Test
+    public void testJMSTopicSource5() throws InterruptedException {
```

Comment:
what are we testing here?

Reply:
fixed 4f8a655

Reply:
Still what you are asserting for "in starting the JMS receiver". I don't see how you can assert connection unavailability with this

Reply:
Here I tested https://github.com/wso2-extensions/siddhi-io-jms/blob/master/component/src/main/java/org/wso2/extension/siddhi/io/jms/source/JMSSource.java#L171

Reply:
Can't you atleast assert for 'Exception in starting the JMS receiver for stream'

Reply:
fixed 001bd05